storage/boltdb/client_test.go

```diff
@@ -4,210 +4,52 @@
 package boltdb
 
 import (
-	"bytes"
-	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-	"go.uber.org/zap"
-	"storj.io/storj/storage"
+	"storj.io/storj/storage/testsuite"
 )
 
-type BoltClientTest struct {
-	*testing.T
-	c storage.KeyValueStore
-}
-
-func NewBoltClientTest(t *testing.T) *BoltClientTest {
-	logger, _ := zap.NewDevelopment()
-	dbName := tempfile()
-
-	c, err := NewClient(logger, dbName, "test_bucket")
-	if err != nil {
-		t.Error("Failed to create test db")
-		panic(err)
-	}
-
-	return &BoltClientTest{
-		T: t,
-		c: c,
-	}
-}
-
-func (bt *BoltClientTest) Close() {
-	bt.c.Close()
-	switch client := bt.c.(type) {
-	case *Client:
-		os.Remove(client.Path)
-	}
-}
-
-func (bt *BoltClientTest) HandleErr(err error, msg string) {
-	bt.Error(msg)
-	if err != nil {
-		panic(err)
-	}
-	panic(msg)
-}
-
-func tempfile() string {
-	f, err := ioutil.TempFile("", "TempBolt-")
-	if err != nil {
-		panic(err)
-	}
-	f.Close()
-	err = os.Remove(f.Name())
+func TestSuite(t *testing.T) {
+	tempdir, err := ioutil.TempDir("", "bolt")
 	if err != nil {
-		panic(err)
-	}
-	return f.Name()
-}
-
-func TestPut(t *testing.T) {
-	bt := NewBoltClientTest(t)
-	defer bt.Close()
-
-	if err := bt.c.Put([]byte("test/path/1"), []byte("pointer1")); err != nil {
-		bt.HandleErr(err, "Failed to save pointer1 to pointers bucket")
+		t.Fatal(err)
 	}
-}
+	defer os.RemoveAll(tempdir)
 
-func TestGet(t *testing.T) {
-	bt := NewBoltClientTest(t)
-	defer bt.Close()
-
-	if err := bt.c.Put([]byte("test/path/1"), []byte("pointer1")); err != nil {
-		bt.HandleErr(err, "Failed to save pointer1 to pointers bucket")
-	}
-
-	retrvValue, err := bt.c.Get([]byte("test/path/1"))
+	dbname := filepath.Join(tempdir, "bolt.db")
+	store, err := New(dbname, "bucket")
 	if err != nil {
-		bt.HandleErr(err, "Failed to get")
-	}
-	if retrvValue.IsZero() {
-		bt.HandleErr(nil, "Failed to get saved test pointer")
-	}
-	if !bytes.Equal(retrvValue, []byte("pointer1")) {
-		bt.HandleErr(nil, "Retrieved pointer was not same as put pointer")
-	}
-
-	// tests Get non-existent path
-	getRes, err := bt.c.Get([]byte("fake/path"))
-
-	// here we are expecting an error for an empty file
-	assert.Nil(t, getRes)
-	assert.True(t, storage.ErrKeyNotFound.Has(err))
-}
-
-func TestDelete(t *testing.T) {
-	bt := NewBoltClientTest(t)
-	defer bt.Close()
-
-	if err := bt.c.Put([]byte("test/path/1"), []byte("pointer1")); err != nil {
-		bt.HandleErr(err, "Failed to save pointer1 to pointers bucket")
+		t.Fatalf("failed to create db: %v", err)
 	}
+	defer func() {
```

Comment:
You can use `defer utils.LogClose(store)` here.

Reply:
`utils.LogClose` logs to the wrong place, tests should be logging into `t`, go test expects that behavior.